src/shogun/mathematics/linalg/linalg.h

```diff
@@ -35,40 +35,40 @@
 #include <shogun/lib/config.h>
 
 /**
- * Just include this file to use in your applications.
+ * Just include this file to use in your applications (in the cpp)
  */
 
-/**
- * This namespace contains all linear algebra specific modules and operations.
- * The rest of the library is available only when the one of the supported
- * backend exists in the local system
- */
-
-#ifdef HAVE_LINALG_LIB
 namespace shogun
 {
 
+/**
+ * This namespace contains all linear algebra specific modules and operations
+ * for which we (may) rely on multiple implementation, either native or
+ * making use of some supported third party external linear algebra libraries.
+ */
 namespace linalg
 {
 
 /**
  * Developer's Note :
- * - Changing the default backend would just require to change it in the following
- *   enum
+ * - Changing the default backend would just require to change it in the
+ *   Backend enum
  * - Please use the same names as HAVE_<BACKEND> macros (e.g. HAVE_EIGEN3)
+ * - Please refer to the developer's wiki (link) for the design and structure
+ *   of internal linalg module
  */
 
 /**
  * @brief
  * All currently supported linear algebra backend libraries, with a default
  * backend, which will be used for all the tasks if any particular backend is
- * not set explicitly via cmake options.
+ * not set explicitly via cmake options. A Native backend is also defined to
+ * accommodate Shogun's native implementation of some operations.
  *
  * The enum defines these backends in order of priority as default backend, as
- * in, first defined one will be used as default
- *
- * Note - Currently EIGEN3 is the default (if its available)
+ * in, first defined one will be used as default.
  *
+ * Note - Currently EIGEN3 is the default (if it is available).
```

Comment:
@iglesias thanks for your comments.
- yes you're right, I'll add some comments on the module names. It is not very clear from those names. IMO redux is for those operations which takes a matrix/vector and returns a scalar, e.g. dot, sum etc. rest generic util things should go to core.
- actually the guard is not required. `HAVE_LINALG_LIB` is defined whenever we have `HAVE_EIGEN3` _or_ `HAVE_VIENNACL` defined. currently there is no cmake option to turn linalg off. I think only `HAVE_EIGEN3` check will suffice there. I'll take care of that.

Thanks. I hope the wiki will be useful :) will finish it off by tonight.